src/hippopt/robot_planning/expressions/quaternion.py

```diff
@@ -0,0 +1,85 @@
+import casadi as cs
+import liecasadi
+
+
+def quaternion_xyzw_normalization(
+    quaternion_xyzw_name: str = "quaternion",
+    options: dict = None,
+    **_,
+) -> cs.Function:
+    options = {} if options is None else options
+    quaternion = cs.MX.sym(quaternion_xyzw_name, 4)
+
+    normalized_quaternion = liecasadi.Quaternion(xyzw=quaternion).normalize()
+
+    return cs.Function(
+        "quaternion_xyzw_normalization",
+        [quaternion],
+        [normalized_quaternion.xyzw],
+        [quaternion_xyzw_name],
+        ["quaternion_normalized"],
+        options,
+    )
+
+
+def quaternion_xyzw_velocity_to_right_trivialized_angular_velocity(
+    quaternion_xyzw_name: str = "quaternion",
+    quaternion_xyzw_velocity_name: str = "quaternion_velocity",
+    options: dict = None,
+    **_,
+) -> cs.Function:
+    options = {} if options is None else options
+    quaternion = cs.MX.sym(quaternion_xyzw_name, 4)
+    quaternion_velocity = cs.MX.sym(quaternion_xyzw_velocity_name, 4)
+
+    q_w = quaternion[3]
+    q_i = quaternion[:3]
+
+    q_dot_w = quaternion_velocity[3]
+    q_dot_i = quaternion_velocity[:3]
```

Comment:
don't be afraid of using `Q̇` in code, it increases readability. I'd suggest to use `Q` for quaternions to avoid collisions with $`\mathbf{q} \in \text{SE}(3) \times \mathbb{R}^n`$.

Reply:
As above, I would avoid using non-ascii characters. Pycharm throws a warning and I don't want to bother it 😁